staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -71,6 +85,8 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 		}
 	}
 
+	remaining := req.TLS.PeerCertificates[0].NotAfter.Sub(time.Now())
+	clientCertificateExpirationGauge.WithLabelValues(req.TLS.PeerCertificates[0].Subject.CommonName).Set(float64(remaining / time.Second))
```

Comment:
I've changed the metric to a histogram to report counts in certain buckets, no detail about source, so the cardinality is low. I'll follow up with an addition to node status so it is possible to easily query nodes about remaining credential lifetime.